### ISSUE TITLE
Change theme colors based on Neovim background setting

### ISSUE THREAD

**Issue by cseelus** [hoob3rt/lualine.nvim#280]
### Requested feature

Make `vim.opt.background:get()` work dynamically inside the theme files, to switch theme colors based on the current Neovim background color setting.

### Motivation

I'm creating a Lualine theme, as part of a Neovim color scheme, that has a light and dark mode. 

Changing Lualine theme colors based on the current Neovim `background` setting, like one can do with regular Neovim color schemes, would really be helpful, to make Lualine adapt in conjunction with the main color scheme.

**Comment by shadmansaleh** [comment]
Can you try with #311 ?
I think it's been resolved with that . There if you do
```lua
require'lualine'.setup {options = {theme = 'theme_name'}}
```
Will reload the theme.

**Comment by cseelus** [comment]
Just checked out `shadmansaleh/lualine.nvim`. Unfortunately this code inside the Lualine theme

```lua
c = colors.Dark
if vim.opt.background:get() == 'light' then
  c = colors.Light
end
```

… which used to at least work until now when starting NeoVIM with a `dark` `background` as well as a `light` one, now always defaults to a dark background, no matter the setting for `background`. Dynamically switching background colors did not work for the Lualine theme before and does not work now.

So kind of a regression, as `vim.opt.background:get() == 'light'` is always evaluated as `false`, I'm afraid.

**Comment by shadmansaleh** [comment]
Strange it's working correctly . Can you try following
commands one by one and see if the theme adapts to background ? Also make sure you don't have a plugin that provides solarised theme for lualine . The builtin one adapts to background (Atlest works for me) .
```vim
:lua require'lualine'.setup{options={theme='solarized'}}
:set background=light
:lua require'lualine'.setup{options={theme='solarized'}}
:set background=dark
:lua require'lualine'.setup{options={theme='solarized'}}
```

**Comment by cseelus** [comment]
You're right, manually setting `lua require'lualine'.setup{options={theme='solarized'}}` (or in case of my theme ` lua require'lualine'.setup{options={theme='tone'}}` after changing the `background` does work.

What does not work unfortunately is the Lualine theme setting itself after changing `background`, while Neovim is still open, so for example changing to a `dark` bg in the evening.

With the new changes, it also stops working with `dark-notify`, which sets the background dynamically, based on the current systems appearance. Before it worked at least on startup, so for example both Neovim and Lualine were set to `dark`, when opened at night and to `light` when opened on daytime, now Lualine is always set to `dark`.

**Comment by shadmansaleh** [comment]
> What does not work unfortunately is the Lualine theme setting itself after changing background, while Neovim is still open, so for example changing to a dark bg in the evening.

Yeah ofcource it's not implemented . And not something I'm thinking of adding to lualine for now. Besides it's awfully simple for anyone to do in their config anyone wanting that. Just add 

```vim
:autocmd OptionSet background lua require'lualine'.setup()
```

> With the new changes, it also stops working with dark-notify

That's definatly odd . What's dark-notify exactly ? is it a plugin. It should work since setting background before loading lualines config during startup works .

I'm thinking it's probably setting background after lualines config is already set . It might have worked before because theme loading used to be scheduled instead of loading immidiatly , I've removed that .dark-notify might have changed background option in between . If this is the case It should be covered by OptionSet auto commad to I think

**Comment by cseelus** [comment]
Yeah, I have `vim.api.nvim_command("autocmd optionset background lua require('lualine').setup()")` in my config for Lualine.

Any particular reason why this auto command can't be added directly, for example to `lualine.nvim/lua/lualine/init.lua`? 
After all, even some of the builtin themes come in a dark and light flavor and changing VIMs `background` on the fly (manually or via a plugin like dark-notify) is not something uncommon.

So Lualine adapting to background changes out of the box, would sure be appreciated by users of the `ayu`, `iceberg`, etc  themes.

Alternatively it might be good to add a hint to use this autocmd to the readme. Could open a PR for that.

**Comment by shadmansaleh** [comment]
Ok I've added this to #311 for now . Changeing background should work out of the box now .
